mParticle-Apple-SDK/Kits/MPKitContainer.mm

```diff
@@ -2306,6 +2306,12 @@ - (void)attemptToLogEventToKit:(id<MPExtensionKitProtocol>)kitRegister kitFilter
                     MPATTAuthorizationStatus status = (MPATTAuthorizationStatus)[parameters[0] unsignedIntValue];
                     NSNumber *timestamp = [parameters[1] isKindOfClass:[NSNumber class]] ? (NSNumber*)parameters[1] : nil;
                     execStatus = [kitRegister.wrapperInstance setATTStatus:status withATTStatusTimestampMillis:timestamp];
+                } else if (selector == @selector(setOptOut:)) {
+                    if (parameters.count == 1 && [parameters[0] boolValue]) {
+                        execStatus = [kitRegister.wrapperInstance setOptOut:true];
+                    } else {
+                        execStatus = [kitRegister.wrapperInstance setOptOut:false];
+                    }
```

Comment:
```suggestion
                    BOOL isOptOut = (parameters.count >= 1 && [parameters[0] boolValue]);
                    execStatus = [kitRegister.wrapperInstance setOptOut:isOptOut];
```

This can be simplified like this, I confirmed this works in Xcode and ran the unit tests as well. 

Technically the `parameters.count >= 1` isn't even required as the subscript implementation of `MPForwardQueueParameters` performs bounds checking, but that's an unusual implementation detail so I like that you have the check here as well to prevent future crashes if that ever changes.